hw/top_earlgrey/data/chip_testplan.hjson

```diff
@@ -3153,5 +3199,42 @@
       stage: V2
       tests: ["chip_sw_pwrmgr_deep_sleep_all_wake_ups"]
     }
+
+    {
+      name: chip_sw_exit_test_locked
+      desc: '''End to end test to ensure rom boot strap can be performed after
+      exiting TEST_LOCKED.
+
+      - Pre-load the device into TEST_LOCKED and TEST_UNLOCKED states.
+      - Advance device to PROD, PROD_END or DEV through LC JTAG request.
+      - Reboot the device and perform boot strap of a simple image, (e.g Hello World).
+      - Ensure boot strap succeeds.
```

Comment:
nit: Would it also make sense to test the ROM enable sequence? I am not sure if there is value in doing so given the amount of work required to perform the OTP write via JTAG.

Reply:
hmmm... i feel like that might be better in the rom e2e test.
Mostly because i have some concerns over how the sequence of `ROM_EXEC_EN` and advance to PROD should be done (i have it in the other doc).

But we can't really emulate that in testrom, because testrom doesn't currently have the ability to shut down the device whenever there is no valid image. 

can you have a look [here](https://docs.google.com/document/d/1qiCqlCicJCFqSpTAuSC_Z3l75NneS2sATlsgg4zKjEg/edit?disco=AAAAgGWcg50) and let me know what you think?